src/main/java/bio/terra/service/dataset/flight/UnlockDatasetStep.java

```diff
@@ -50,9 +62,12 @@ public StepResult doStep(FlightContext context) {
                 "Expected dataset id to either be passed in or in the working map."));
       }
     }
+    if (lockName == null) {
+      lockName = context.getFlightId();
+    }
 
     try {
-      datasetService.unlock(datasetId, context.getFlightId(), sharedLock);
+      datasetService.unlock(datasetId, lockName, sharedLock);
```

Comment:
This behavior may be acceptable, but I wanted to flag it!

Right now, `datasetService.unlock` ignores whether or not the lock was actually removed. For example, if the dataset were locked by flight `abc-123` and we tried to remove lock `def-456`, the lock would be unchanged, the flight would succeed, and we'd journal that we removed the lock.